File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    public List<TbBrand> findAll();

    /**
     * 品牌列表页面展示，带分页
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 品牌添加
     * @param tbBrand
     * @return
     */
    public void add(TbBrand tbBrand);

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    public TbBrand findById(long id);

    /**
     * 更新品牌
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);

    /**
     * 删除品牌功能
     */
    public void delete(Long[] ids);

    /**
     * 根据输入品牌名称和首字母进行模糊查询品牌功能
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult search(TbBrand tbBrand,int pageNum,int pageSize);

    /**
     * 获取品牌下拉选项
     * @return
     */
    public List<Map> brandOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//service层
app.service("brandService",function($http) {
     this.findAll = function() {
         return $http.get("../brand/findAll.do");
     }
     this.findPage = function(page,size) {
         return $http.get("../brand/findPage.do?page="+page+"&size="+size);
     }
     this.findById = function(id) {
         return $http.get("../brand/findById.do?id="+id);
     }
     this.add = function(entity) {
         return $http.post("../brand/add.do",entity);
     }
     this.update = function(entity) {
         return $http.post("../brand/update.do",entity);
     }
     this.del = function(ids) {
         return $http.get("../brand/delete.do?ids="+ids);
     }
     this.search = function(page,size,searchBrand) {
         return $http.post("../brand/search.do?page="+page+"&size="+size,searchBrand);
     }

     this.brandOptionList = function () {
         return $http.get("../brand/brandOptionList.do");
     }
 });